lib/pf/scan.pm

```diff
@@ -342,6 +343,59 @@ sub toString {
     return $self->{'_id'};
 }
 
+=head2 matchCategory
+
+Check if the category matches the configuration of the scanner
+
+=cut
+
+sub matchCategory {
+    my ($self, $node_attributes) = @_;
+    my $category = [split(/\s*,\s*/, $self->{_categories})];
+    my $node_cat = $node_attributes->{'category'};
+
+    get_logger->debug( sub { "Tring to match the role '$node_cat' against " . join(",", @$category) });
+    # validating that the node is under the proper category for provisioner
+    return @$category == 0 || any { $_ eq $node_cat } @$category;
+}
+
+=head2 matchOS
+
+Check if the OS matches the configuration of the scanner
+
+=cut
+
+sub matchOS {
+    my ($self, $node_attributes) = @_;
+    my @oses = @{$self->{_oses} || []};
+
+    my $fingerbank_info = pf::node::fingerbank_info($node_attributes->{mac}, $node_attributes);
```

Comment:
Place this after the check for no oses to avoid unnecessary lookups if there are no oses defined.